json-endpoints/jetpack/class.jetpack-json-api-sync-endpoint.php

```diff
@@ -234,3 +241,144 @@ protected function result() {
 		);
 	}
 }
+
+
+class Jetpack_JSON_API_Sync_Checkout_Endpoint extends Jetpack_JSON_API_Sync_Endpoint {
+	protected function result() {
+		$args = $this->query_args();
+		$queue_name = $this->validate_queue( $args['queue'] );
+
+		if ( is_wp_error( $queue_name ) ){
+			return $queue_name;
+		}
+
+		if ( is_int( $args[ 'number_of_items' ] ) && (int) $args[ 'number_of_items' ] < 1 ) {
+			return  new WP_Error( 'invalid_number_of_items', 'Number of items needs to be in integer and should be larger that is larger then 0', 400 );
+		}
+		require_once JETPACK__PLUGIN_DIR . 'sync/class.jetpack-sync-sender.php';
+		require_once JETPACK__PLUGIN_DIR . 'sync/class.jetpack-sync-queue.php';
+		require_once JETPACK__PLUGIN_DIR . 'sync/class.jetpack-sync-json-deflate-array-codec.php';
+
+		$queue = new Jetpack_Sync_Queue( $queue_name );
+		$codec = new Jetpack_Sync_JSON_Deflate_Array_Codec();
+
+		// We need the sender to set up all the before send module actions...
+		$sender = Jetpack_Sync_Sender::get_instance();
+		
+		$skipped_items_ids = array();
+
+		// let's delete the checkin state
+		if ( isset( $args['force'] ) && $args['force'] ) {
+			$queue->force_checkin();
+		}
+
+		if ( isset( $args['show_errors'] ) && $args['show_errors'] ) {
+			error_reporting( E_ERROR | E_WARNING | E_PARSE );
+		}
```

Comment:
Yeah I am not sure how this will work. Still need to test this. But the idea is that we would get back errors that might be helpful to us. Not sure this is even possible. We might have to take it out if it doesn't really work as expected. 

The I was planning to test it is I would create a plugin that calls a shortcode that calls a function that doesn't exists. and add the shortcode to a post.